src/wolff_producer.erl

```diff
@@ -279,6 +313,27 @@ terminate(_, #{replayq := Q}) ->
 terminate(_, _) ->
   ok.
 
+aggregate_send_calls(BatchCalls, Limit) ->
+  aggregate_send_calls(BatchCalls, Limit, {[], 0, 0}, []).
+
+aggregate_send_calls([], _, {CallsAcc, Cnt, Size}, Res) ->
+  Res1 = [{lists:reverse(CallsAcc), Cnt, Size} | Res],
+  lists:reverse(Res1);
+aggregate_send_calls([?SEND_REQ(_, Batch, _) = Call | Calls], Limit, {CallsAcc, Cnt, Size}, Res) ->
+  case Size + batch_bytes(Batch) of
+    Size1 when Size1 > Limit, Size == 0 ->
+      Res1 = [{[Call], Cnt + 1, Size1} | Res],
+      aggregate_send_calls(Calls, Limit, {[], 0, 0}, Res1);
+    Size1 when Size1 > Limit ->
+      Res1 = [{lists:reverse(CallsAcc), Cnt, Size} | Res],
+      aggregate_send_calls([Call | Calls], Limit, {[], 0, 0}, Res1);
+    Size1 ->
+      aggregate_send_calls(Calls, Limit, {[Call | CallsAcc], Cnt + 1, Size1}, Res)
+  end.
+
+producer_name(ClientId, Partition) ->
+  list_to_atom(lists:flatten(io_lib:format("~s-~p", [ClientId, Partition]))).
+
```

Comment:
This is just for adding unique names to the ETS tables, it would be easier for troubleshooting.

Reply:
does it have to be a named table ?

Reply:
Actually it is not a named table, I didn't pass a `named_table` option to the `ets:new/2`. But if we pass the `ets:new/2` a meaningful name, it would be easier for us to troubleshoot the system using `ets:i()`.